benchmark/rocblas/blas2/gbmv.cpp

```diff
@@ -0,0 +1,217 @@
+
+    CHECK_ROCBLAS_STATUS(
+        rocblas_set_pointer_mode(rb_handle, rocblas_pointer_mode_host));
+
+#ifdef BLAS_VERIFY_BENCHMARK
+    // Reference gbmv
+    std::vector<scalar_t> v_y_ref = v_y;
+    reference_blas::gbmv(t_str, m, n, kl, ku, alpha, m_a.data(), lda,
+                         v_x.data(), incX, beta, v_y_ref.data(), incY);
+
+    // Rocblas verification gbmv
+    std::vector<scalar_t> v_y_temp = v_y;
+    {
+      // Temp result on device (copied back to Host upon destruction)
+      blas_benchmark::utils::HIPVector<scalar_t, true> y_temp_gpu(
+          ylen, v_y_temp.data());
+      // rocBLAS function call
+      rocblas_gbmv_f<scalar_t>(rb_handle, transA, m, n, kl, ku, &alpha, m_a_gpu,
+                               lda, v_x_gpu, incX, &beta, y_temp_gpu, incY);
+    }
+
+    std::ostringstream err_stream;
+    if (!utils::compare_vectors(v_y_temp, v_y_ref, err_stream, "")) {
+      const std::string& err_str = err_stream.str();
+      state.SkipWithError(err_str.c_str());
+      *success = false;
+    };
+#endif
+
+    auto blas_warmup = [&]() -> void {
+      rocblas_gbmv_f<scalar_t>(rb_handle, transA, m, n, kl, ku, &alpha, m_a_gpu,
+                               lda, v_x_gpu, incX, &beta, v_y_gpu, incY);
+      return;
+    };
+
+    hipEvent_t start, stop;
+    CHECK_HIP_ERROR(hipEventCreate(&start));
+    CHECK_HIP_ERROR(hipEventCreate(&stop));
+
+    auto blas_method_def = [&]() -> std::vector<hipEvent_t> {
+      CHECK_HIP_ERROR(hipEventRecord(start, NULL));
+      rocblas_gbmv_f<scalar_t>(rb_handle, transA, m, n, kl, ku, &alpha, m_a_gpu,
+                               lda, v_x_gpu, incX, &beta, v_y_gpu, incY);
+      CHECK_HIP_ERROR(hipEventRecord(stop, NULL));
+      CHECK_HIP_ERROR(hipEventSynchronize(stop));
+      return std::vector{start, stop};
+    };
+
+    // Warmup
+    blas_benchmark::utils::warmup(blas_warmup);
+    CHECK_HIP_ERROR(hipStreamSynchronize(NULL));
+
+    blas_benchmark::utils::init_counters(state);
+
+    // Measure
+    for (auto _ : state) {
+      // Run
+      std::tuple<double, double> times =
+          blas_benchmark::utils::timef_hip(blas_method_def);
+
+      // Report
+      blas_benchmark::utils::update_counters(state, times);
+    }
+
+    state.SetItemsProcessed(state.iterations() * nflops_tot);
+
+    blas_benchmark::utils::calc_avg_counters(state);
+
+    CHECK_HIP_ERROR(hipEventDestroy(start));
+    CHECK_HIP_ERROR(hipEventDestroy(stop));
+  }  // release device memory via utils::DeviceVector destructors
+}
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args, rocblas_handle& rb_handle,
+                        bool* success) {
+  auto gbmv_params = blas_benchmark::utils::get_gbmv_params<scalar_t>(args);
+
+  for (auto p : gbmv_params) {
+    std::string ts;
+    index_t m, n, kl, ku;
+    scalar_t alpha, beta;
+    std::tie(ts, m, n, kl, ku, alpha, beta) = p;
+    int t = static_cast<int>(blas_benchmark::utils::to_transpose_enum(ts));
```

Comment:
I did the same in `cublas` PR, so this is a question for both of us. 
`ts` is casted to `int` passed to the lambda and casted back to string in the `run` function(line 51), why don't we just pass it as a string or a char?

Reply:
Yes indeed. As discussed, this will be subject of another PR.